### ISSUE TITLE
comment-on-pr-number ignored unless event type is `pull_request`

### ISSUE THREAD

**Issue by bennettp123** [pulumi/actions#802]
### What happened?

I'm trying to use `comment-on-pr-number` within a push event to ensure PR comments work as expected. However, it doesn't attempt to comment on the PR

### Steps to reproduce

1. Add the workflow below
2. Merge a PR into the `main` branch

```yaml
# main.yml

on:
  push:
    - main

jobs:
  pulumi:
    steps:
    - uses: jwalton/gh-find-current-pr@v1
        id: pr
        with:
          state: all
      - uses: pulumi/actions@v3
        with:
          command: up
          stack-name: ${{ inputs.stack-name }}
          comment-on-pr-number: ${{ steps.pr.outputs.pr }}
          edit-pr-comment: false
```

### Expected Behavior

* `pulumi up` action is triggered
* output is posted to the merged PR as a comment

### Actual Behavior

* `pulumi up` action is triggered
* status is not added to the merged PR

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by RobbieMcKinstry** [comment]
Hi @bennettp123 thanks for opening this issue. Am I understanding correctly?
The issue is that if you have an already open PR, no comment appears.

**Comment by bennettp123** [comment]
> Hi @bennettp123 thanks for opening this issue. Am I understanding correctly?
> The issue is that if you have an already open PR, no comment appears.

No, the open PR is working fine—that's omitted that from the example, but when using `on: {pull_request: {target: [main]}}`, and it creates a comment in the PR—that works fine. I'm also able to use `on: {pull_request: {target: [main], types: [closed]}}`—that works fine too.

However, I'd like to switch to `on: {push: [main]}`, while still being able to add a comment to the PR. I figured I could use `jwalton/gh-find-current-pr@v1` with `comment-on-pr-number` to tell it which PR to comment on—which should work fine, but it's currently not working.

To summarise:
* Create a new PR &larr; working fine
* Push to an existing PR &larr; working fine (I assume—haven't tested)
* Merge a PR into the main branch:
    * `pull_request` trigger &larr; working fine
    * `push` trigger:
        * `comment-on-pr-number` is not set &larr; working fine
        * `comment-on-pr-number` is set to a valid PR number: no comment is created &larr; not working

**Comment by RobbieMcKinstry** [comment]
Oh wow, that's was an extremely helpful explanation! Thank you very much! If you get a chance to add a test to your PR, I'll try to turn it around quickly. With the holiday approaching, it might slip to the new year, but I'll keep my eyes open for it 

Thank you very much for your contribution, both with your thorough explanation and your initial pass at a solution! ❤️
